src/Microsoft.Framework.DesignTimeHost/DesignTimeAssemblyLoadContextFactory.cs

```diff
@@ -62,9 +57,12 @@ private IAssemblyLoadContextFactory GetRuntimeFactory(IServiceProvider servicePr
 
                 // Add a cache dependency on restore complete to reevaluate dependencies
                 ctx.Monitor(_namedDependencyProvider.GetNamedDependency(_project.Name + "_Dependencies"));
-
-                return new AssemblyLoadContextFactory(applicationHostContext.ServiceProvider);
+                
+                return applicationHostContext;
             });
+            
+            var factory = new AssemblyLoadContextFactory(appHostContext.ServiceProvider);
```

Comment:
For more background: If the `ApplicationHostContext` is cached why did caching the `AssemblyLoadContextFactory` based on the still cached `ApplicationHostContext` break the scenario? Isn't that assuming the `ServiceProvider` gets updated (since it can't be set)? From my understanding of `IServiceProvider` that shouldn't be possible. Definitely missing something here.